### ISSUE TITLE
occasional segfault and crash 

### ISSUE THREAD

**Issue by stefson** [Return-To-The-Roots/s25client#745]
The backtrace of this does look pretty much like the one from #740, but is another issue from my point of view. I attached the replay, plus the backtrace as plain text, am at 02770d57f2ae8432139edc3e6f00cc3a1a54c723

[replay.zip](https://github.com/Return-To-The-Roots/s25client/files/1463521/replay.zip)

```
Thread 1 "s25client" received signal SIGSEGV, Segmentation fault.
glArchivItem_Bitmap_Player::DrawFull (this=0x0, dst=..., color=4294967295, player_color=4294967295)
    at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/ogl/glArchivItem_Bitmap_Player.cpp:39
39	/var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/ogl/glArchivItem_Bitmap_Player.cpp: Datei oder Verzeichnis nicht gefunden.
(gdb) bt
#0  glArchivItem_Bitmap_Player::DrawFull (this=0x0, dst=..., color=4294967295, player_color=4294967295)
    at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/ogl/glArchivItem_Bitmap_Player.cpp:39
#1  0x00000000006c2d8b in noSign::Draw (this=0x566b760, drawPt=...) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/nodeObjs/noSign.cpp:62
#2  0x00000000006eef4f in GameWorldView::DrawObject (this=this@entry=0x8d42838, pt=..., curPos=...)
    at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/world/GameWorldView.cpp:480
#3  0x00000000006f1cdb in GameWorldView::Draw (this=this@entry=0x8d42838, rb=..., selected=..., drawMouse=<optimized out>, water=water@entry=0x7fffffffd36c)
    at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/world/GameWorldView.cpp:178
#4  0x0000000000761ce2 in dskGameInterface::Run (this=this@entry=0x8d42610) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/desktops/dskGameInterface.cpp:866
#5  0x0000000000761e8e in dskGameInterface::Msg_PaintBefore (this=0x8d42610) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/desktops/dskGameInterface.cpp:255
#6  0x00000000005d4e20 in WindowManager::Draw (this=0xbfff90) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/WindowManager.cpp:70
#7  0x000000000058056a in GameManager::Run (this=0xbff690) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/GameManager.cpp:168
#8  0x000000000055f41d in RunProgram (options=...) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/s25client/s25client.cpp:375
#9  0x000000000051c54d in main (argc=1, argv=0x7fffffffdc18) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/s25client/s25client.cpp:433
(gdb)

**Comment by Flamefire** [comment]
Thank you. For draw related crashes please try to remember where it crashed (what area of the map you viewed) that would speed things up

**Comment by stefson** [comment]
Okay, I will provide a screenshot if this should ever happen again.

**Comment by Flamefire** [comment]
Don't need to be a screenshot as this would be hard on a crash. Just something: View was next to reds HQ or so. Fixing bugs is hard so I really appreciate any minute saved :)

**Comment by stefson** [comment]
I was near my own headquarter, mostly building stuff either norh or south of it. Screenshot should be possible if having the game within the gnu debugger, because it halts abruptly when the segfault happens. If you can try to fix #744 first, not to wrry about savegames would make testing a lot easier.

**Comment by stefson** [comment]
new session with debugger, the replay: [2.017-11-11_18-37-21.rpl.zip](https://github.com/Return-To-The-Roots/s25client/files/1463970/2.017-11-11_18-37-21.rpl.zip)

and this is the frame where the debugger stopped the binary for debugging

 ![segfault](https://user-images.githubusercontent.com/14232936/32692129-cd5979ce-c712-11e7-9d35-87cd8873586d.png)

This is the output from the debugger (same bug I guess)
```
Thread 1 "s25client" received signal SIGSEGV, Segmentation fault.
glArchivItem_Bitmap_Player::DrawFull (this=0x0, dst=..., color=4294967295, 
    player_color=4294967295)
    at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/ogl/glArchivItem_Bitmap_Player.cpp:39
39	/var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/ogl/glArchivItem_Bitmap_Player.cpp: Datei oder Verzeichnis nicht gefunden.
(gdb) bt
#0  glArchivItem_Bitmap_Player::DrawFull (this=0x0, dst=..., color=4294967295, player_color=4294967295)
    at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/ogl/glArchivItem_Bitmap_Player.cpp:39
#1  0x00000000006c2ffb in noSign::Draw (this=0xa9c94f0, drawPt=...) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/nodeObjs/noSign.cpp:62
#2  0x00000000006ef1bf in GameWorldView::DrawObject (this=this@entry=0x8d42c08, pt=..., curPos=...)
    at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/world/GameWorldView.cpp:480
#3  0x00000000006f1f4b in GameWorldView::Draw (this=this@entry=0x8d42c08, rb=..., selected=..., drawMouse=<optimized out>, water=water@entry=0x7fffffffd36c)
    at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/world/GameWorldView.cpp:178
#4  0x0000000000761f52 in dskGameInterface::Run (this=this@entry=0x8d429e0) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/desktops/dskGameInterface.cpp:866
#5  0x00000000007620fe in dskGameInterface::Msg_PaintBefore (this=0x8d429e0) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/desktops/dskGameInterface.cpp:255
#6  0x00000000005d4ec0 in WindowManager::Draw (this=0xc00550) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/WindowManager.cpp:70
#7  0x000000000058060a in GameManager::Run (this=0xbff690) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/GameManager.cpp:168
#8  0x000000000055f41d in RunProgram (options=...) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/s25client/s25client.cpp:375
#9  0x000000000051c54d in main (argc=1, argv=0x7fffffffdc18) at /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/src/s25client/s25client.cpp:433
